Source/WebKit/Platform/IPC/Connection.cpp

```diff
@@ -75,11 +75,22 @@ struct Connection::WaitForMessageState {
     std::unique_ptr<Decoder> decoder;
 };
 
-class Connection::SyncMessageState {
+Lock s_syncStateLock;
+
+class Connection::SyncMessageStateImpl : public Connection::SyncMessageState {
 public:
-    static SyncMessageState& singleton();
+    static Ref<Connection::SyncMessageState> forCurrentThread(SerialFunctionDispatcher*);
```

Comment:
Your lookup function cannot be
"SyncMessageState for current thread serial function dispatcher".
Name doesn't make sense, it does not look anything up for current thread.

So the lookup function is 
"SyncMessageState for this serial function dispatcher."
So I'd name it "SyncMessageState::get(SerialFunctionDispatcher&)".